app/code/Magento/NewRelicReporting/Plugin/CommandPlugin.php

```diff
@@ -46,10 +54,39 @@ public function __construct(
      */
     public function beforeRun(Command $command, ...$args)
     {
-        $this->newRelicWrapper->setTransactionName(
-            sprintf('CLI %s', $command->getName())
-        );
+        if (!$this->isCommandSkipped($command)) {
+            $this->newRelicWrapper->setTransactionName(
+                sprintf('CLI %s', $command->getName())
+            );
+        }
 
         return $args;
     }
+
+    /**
+     * Ends transaction after successful run
+     *
+     * @param Command $command
+     * @param int $result
+     * @return int The command exit code
+     */
+    public function afterRun(Command $command, $result)
+    {
+        if (!$this->isCommandSkipped($command)) {
+            $this->newRelicWrapper->endTransaction();
+        }
+
+        return $result;
+    }
+
+    /**
+     * Determines whether the Command is declared to be skipped
+     *
+     * @param Command $command
+     * @return bool
+     */
+    private function isCommandSkipped(Command $command): bool
+    {
+        return in_array($command->getName(), $this->skipCommands);
```

Comment:
Looks like this code will be executed quite often. In array actually is quite slow, could you replace it to finding in map?
Related https://github.com/magento/magento-coding-standard/issues/143

Reply:
This part is run only for bin/magento commands execution, however - I changed it to avoid in_array